Use the packaged tinyxml2 instead of compiling the bundled copy.
Fedora also devendors tinyxml2 in GPUArchitectureGenerator.

--- a/GPUArchitectureGenerator/CMakeLists.txt
+++ b/GPUArchitectureGenerator/CMakeLists.txt
@@ -1,13 +1,15 @@
 # Copyright Advanced Micro Devices, Inc., or its affiliates.
 # SPDX-License-Identifier:  MIT
+
+find_package(tinyxml2 CONFIG REQUIRED)
 
 add_library(isa-spec-manager OBJECT)
 target_sources(isa-spec-manager
     PRIVATE
         "${CMAKE_CURRENT_SOURCE_DIR}/isa_spec_manager/isa_xml_reader.cpp"
-        "${CMAKE_CURRENT_SOURCE_DIR}/isa_spec_manager/tinyxml2.cpp"
         "${CMAKE_CURRENT_SOURCE_DIR}/isa_spec_manager/amdisa_utility.cpp"
 )
+target_link_libraries(isa-spec-manager PUBLIC tinyxml2::tinyxml2)
 add_library(rocroller::isa-spec-manager ALIAS isa-spec-manager)
 target_include_directories(isa-spec-manager
     PUBLIC
